go.mod

```diff
@@ -1,29 +1,28 @@
 module github.com/operator-framework/kubectl-operator
 
-go 1.23.0
+go 1.23.4
 
-toolchain go1.23.4
+toolchain go1.23.6
```

Comment:
I think we need to keep it as it is all projects to make the life easier
go 1.23.0 and go1.23.4

Reply:
Looks we don't specify it in operator-controller - maybe we don't need to do that here as well?

Reply:
+1 to remove it. cc @tmshort .

Reply:
We want to avoid anything above 1.23.4, try explicitly using 1.23.4, and re-tidy this go.mod

Reply:
(i.e. download go 1.23.4, and use that for any `go` operations)

Reply:
did `go get go@1.23.4`, reverted toolchain change and `go tidied` - the result is go 1.23.4 with no explicit toolchain line (probably as per `For example, a go.mod that says go 1.21.0 with no toolchain line is interpreted as if it had a toolchain go1.21.0 line.` - https://go.dev/doc/toolchain#config)

we should be good to go 🎊